### PR TITLE
Add pruning interval to PyTorchIgnitePruningHandler.

### PR DESCRIPTION
**Pull request by VladSkripniuk**
Address #1136 .

### CODE REVIEW

File: optuna/integration/pytorch_ignite.py [hunk 1]
```diff
@@ -30,18 +30,26 @@ class PyTorchIgnitePruningHandler(object):
...
 
         self._trial = trial
         self._metric = metric
         self._trainer = trainer
+        self._interval = interval
 
     def __call__(self, engine):
         # type: (Engine) -> None
 
+        if self._trainer.state.epoch % self._interval != 0:
```
Comment (keisuke-umezawa) [inline optuna/integration/pytorch_ignite.py:50]: Just a question:
Why it is not `if (self._trainer.state.epoch + 1) % self._interval != 0:`? The reason is that  `__call__` is called after `self._trainer.state.epoch` is incremented?

In that case, `message = "Trial was pruned at {} epoch.".format(self._trainer.state.epoch)` should be below?

```
message = "Trial was pruned at {} epoch.".format(self._trainer.state.epoch - 1)
```

Comment (VladSkripniuk) [reply optuna/integration/pytorch_ignite.py:50]: Pytorch Ignite uses 1-based counting for epochs and iterations ([link](https://github.com/pytorch/ignite/blob/1eddc7176264c03607511467550114a69cb502b4/ignite/engine/events.py#L275)). So `if (self._trainer.state.epoch + 1) % self._interval != 0:` would call pruning after `self._interval-1` epochs. `if self._trainer.state.epoch % self._interval != 0:` would call pruning callback after running first `self._interval` epochs, which is more reasonable in my opinion. 

`self._trainer.state.epoch` is incremented at the very beginning of an epoch ([link](https://github.com/pytorch/ignite/blob/1eddc7176264c03607511467550114a69cb502b4/ignite/engine/engine.py#L675)) so I think it's not necessary to alter the `message`.

Comment (keisuke-umezawa) [reply optuna/integration/pytorch_ignite.py:50]: Thank you for the explanation!
Could you add this information as a comment? ``Pytorch Ignite uses 1-based counting for epochs and iterations (link)``.

**Comment by Crissman** [comment]
@VladSkripniuk This PR is stale and has conflicts with master, so closing for now. If there's further interest, we may want to take a look at https://github.com/optuna/optuna/pull/1161#issuecomment-625598685

Please re-open if you want to proceed and resolve the conflicts.

**Pull request closed without merging**